plugins/gimp/file-jxl-save.cc

```diff
@@ -3,130 +3,813 @@
+  gtk_widget_show(frame_advanced);
+
+  gtk_widget_set_sensitive(frame_advanced, false);
+
+  vbox = gtk_vbox_new(false, 6);
+  gtk_container_set_border_width(GTK_CONTAINER(vbox), 6);
+  gtk_container_add(GTK_CONTAINER(frame_advanced), vbox);
+  gtk_widget_show(vbox);
+
+  advanced_opts.push_back(frame_advanced);
+
+  // lossless convenience checkbox
+  static gchar lossless_help[] =
+      "This is a convenience option that sets distance to 0.";
+  toggle = gtk_check_button_new_with_label("Lossless mode (very slow)");
+  gimp_help_set_help_data(toggle, lossless_help, nullptr);
+  gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(toggle),
+                               jxl_save_opts.lossless);
+  gtk_box_pack_start(GTK_BOX(vbox), toggle, false, false, 0);
+  gtk_widget_show(toggle);
+
+  GtkAdjustment* adjustments[] = {entry_distance, entry_quality};
+
+  g_signal_connect(toggle, "toggled", G_CALLBACK(gui_on_change_lossless),
+                   adjustments);
+
+  advanced_opts.push_back(toggle);
+
+  // save codestream format
+  static gchar codestream_help[] =
+      "Save in JPEG XL codestream format.  This format does not support "
+      "metadata.";
+  toggle = gtk_check_button_new_with_label("Codestream Format");
+  gimp_help_set_help_data(toggle, codestream_help, nullptr);
+  gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(toggle),
+                               !jxl_save_opts.use_container);
+  gtk_box_pack_start(GTK_BOX(vbox), toggle, false, false, 0);
+  gtk_widget_show(toggle);
+
+  g_signal_connect(toggle, "toggled", G_CALLBACK(gui_on_change_codestream),
+                   nullptr);
+
+  advanced_opts.push_back(toggle);
+
+  // uses_original_profile
+  static gchar uses_original_profile_help[] =
+      "Prevents conversion to XYB colorspace.  "
+      "File sizes are approximately doubled.  "
+      "This option is not recommended.";
+  toggle = gtk_check_button_new_with_label("Use Original Color Profile");
+  gimp_help_set_help_data(toggle, uses_original_profile_help, nullptr);
+  gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(toggle),
+                               jxl_save_opts.basic_info.uses_original_profile);
+  gtk_box_pack_start(GTK_BOX(vbox), toggle, false, false, 0);
+  gtk_widget_show(toggle);
+
+  g_signal_connect(toggle, "toggled",
+                   G_CALLBACK(gui_on_change_uses_original_profile), nullptr);
+
+  advanced_opts.push_back(toggle);
+
+  // Save Exif Metadata
+  toggle = gtk_check_button_new_with_label("Save Exif Metadata");
+  gimp_help_set_help_data(
+      toggle, "This setting is not yet available in the API.", nullptr);
+  gtk_box_pack_start(GTK_BOX(vbox), toggle, false, false, 0);
+  gtk_widget_set_sensitive(toggle, false);
+  gtk_widget_show(toggle);
+
+  advanced_opts.push_back(toggle);
+
+  // Encoding Effort/Speed
+  static gchar effort_help[] =
+      "Encoding Effort: Higher number is more effort (slower).\n\tDefault = 7.";
+  frame = gtk_frame_new("Encoding Effort/Speed");
```

Comment:
Just from the option name, I'd consider "effort/speed" somewhat confusing and I'd just leave Effort there. WDYT?

Reply:
I was planning to change it to just "Encoding Effort", as you suggest.  But I thought all the force pushes may have been delaying review, so I was holding off on it.